src/lib.rs

```diff
@@ -34,17 +35,52 @@ const ENV_NAME: &str = "MITOSIS_CONTENT_PROCESS_ID";
 /// It is not unsafe to omit this function, however `mitosis::spawn`
 /// may lead to unexpected behavior.
 pub fn init() {
-    if env::args().len() != 1 {
-        return;
-    }
-    if let Ok(token) = std::env::var(ENV_NAME) {
+    if let Ok(token) = env::var(ENV_NAME) {
         bootstrap_ipc(token);
     }
     // Clear environment variable so processes spawned from the `spawn` closure can
     // themselves be using `mitosis`
     std::env::remove_var(ENV_NAME);
 }
 
+lazy_static::lazy_static! {
+    // This is necessary because it's not possible to downgrade a `RwLockWriteGuard` to
+    // a `RwLockReadGuard` without unlocking. So if we write to make sure that tests using
+    // `init_test` do not interfere with each other (e.g. by overwriting the `TEST_HOOK`),
+    // we need to block for the entire execution of the test.
+    static ref PROTECT_WRITING_TO_TEST_HOOK: Mutex<()> = Mutex::new(());
+    static ref TEST_HOOK: RwLock<Option<&'static str>> = RwLock::new(None);
+}
+
+/// Initialize `mitosis` within a `#[test]`. You need to specify the name of the
+/// test (including its entire module path), because this calls the test binary
+/// with the test's name as the argument, just like `cargo test your::test::name`
+/// does. Technically you can specify an test name that runs `init`, so for simplicity
+/// you may want to include a `#[test] fn mitosis() { init() }` in your test suite and
```

Comment:
We could also just require the user to provide such a test. If it doesn't exist, `.join()` will just hang forever. That would simplify the API